strongbox-commons/src/main/java/org/carlspring/strongbox/io/ReplacingInputStream.java

```diff
@@ -0,0 +1,102 @@
+package org.carlspring.strongbox.io;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+
+public class ReplacingInputStream extends FilterInputStream {
```

Comment:
Would you mind adding a short Javadoc explaining when it's used and what we need it for? Yeah, it's clear that it replaces some data in an `InputStream`, but not clear at all for what purpose, or cases.